Feature: Add libwayland-egl FFI bindings

cmack opened the pull request:
This is chained on #2 but I'm happy to rebase

These are pretty simple bindings since libwayland-egl's interface itself is so small. I hope it's in scope of this project since this lib appears to be included with wayland dev libraries.

cmack committed:
Add FFI bindings to libwayland-egl

Allows the creation of hardware accelerated EGL buffers that can be attached
to Wayland surfaces.

andrewsoutar commented:
My main concern with this is in adding a runtime dependency on `libwayland-egl`. I typically build applications as images, and `cl-wayland-client` is designed so that, when built into an image, it doesn't have any dependencies on the wayland development files and the only dependency is `libwayland-client` itself. (I'm pretty sure I also got `concatenate-source-op` working, so that you can distribute the application as a `.lisp` file with the protocol-definition-generated Lisp code already built-in.) For that reason, I don't want to add a hard runtime dep on `libwayland-egl`for any user of the library. (On my machine - Fedora 36 - it gets pulled in by GTK3 and wayland-devel, but not by `libwayland-client` itself.)

With that said, as it's currently implemented, this won't automatically get pulled in by users of `#:com.andrewsoutar.cl-wayland-client` - it will need to be loaded explicitly as `#:com.andrewsoutar.cl-wayland-client/egl-core`. If that's fine with you, it seems like a fine compromise to me.

Just a couple other things for you to think about:
- Why `egl-core` and not just `egl`? Calling it `egl.lisp` (a) is a bit shorter (`com.andrewsoutar.cl-wayland-client` is annoyingly long enough - I take full responsibility for that one) and (b) matches the library (`libwayland-egl`, not `libwayland-egl-core`).
- Especially if this is going to be an optional module, we should probably at least mention it in the README

cmack commented:
I don't have much experience with delivering lisp images. I agree I would like any runtime dependency to be optional and at most the binary `.so` file, not all the dev files. If that's not how it is currently, do you know how it could be corrected? Else, I'll give a look around at other packages like `cl-opengl` to see how they ensure the lisp core stays linked to the `.so` files

> Why `egl-core` and not just `egl`?

I wasn't sure if this was even the right project to put it or if you were interested in pull requests so I tried following convention. I like `egl` better too. I'll change it.

I'll update the README as well

cmack committed:
Add an EGL example similar to the wl-shm example

Draws a checkerboard using EGL and OpenGL (very old style)

cmack committed:
Add information about EGL buffers to the README

cmack commented:
@andrewsoutar sorry for the delay. I've updated the README and added an example. I put the example in its own file since it seems like the README would get unwieldy with another long example block. I also moved the package to `egl` instead of `egl-core`

cmack commented:
@andrewsoutar is there anything else I should address or make better? I'm still committed to getting this to a merge-able state if you have any feedback.
